pkg/system/xattrs_linux.go

```diff
@@ -10,24 +10,23 @@ func Lgetxattr(path string, attr string) ([]byte, error) {
 	dest := make([]byte, 128)
 	sz, errno := unix.Lgetxattr(path, attr, dest)
 
-	switch {
-	case errno == unix.ENODATA:
-		return nil, nil
-	case errno == unix.ERANGE:
-		// 128 byte array might just not be good enough. A dummy buffer is used
-		// to get the real size of the xattrs on disk
+	for errno == unix.ERANGE {
+		// Buffer too small, get the real size first
```

Comment:
Perhaps keep info about the dummy buffer, or would that be obvious?

```suggestion
		// Buffer too small, use a dummy buffer to get the real size of the xattrs on disk
```

Reply:
Makes sense; patch updated. The key here is we're using a buffer with size of 0.